BAPromiseTests/PromiseBlocksTests.swift

```diff
@@ -0,0 +1,183 @@
+            doneExpect.fulfill()
+        }
+        
+        let observedExpect = XCTestExpectation()
+        promiseBlock.observed = { obj in
+            XCTAssertEqual(obj as! Int, callObj)
+            XCTAssertEqual(DispatchQueue.getSpecific(key: key), "testCallNonErrorOnQueue")
+            observedExpect.fulfill()
+        }
+        
+        promiseBlock.rejected = { obj in
+            XCTFail("unexpected rejection")
+        }
+        
+        let alwaysExpect = XCTestExpectation()
+        promiseBlock.always = {
+            XCTAssertEqual(DispatchQueue.getSpecific(key: key), "testCallNonErrorOnQueue")
+            alwaysExpect.fulfill()
+        }
+        
+        promiseBlock.call(with: .success(callObj))
+        
+        self.wait(for: [doneExpect, observedExpect, alwaysExpect], timeout: 10)
+    }
+    
+    func testCallError() {
+        let promiseBlock = Promise<Void>.PromiseBlock(cancellationToken: PromiseCancelToken())
+        let callObj = DummyError.dummy
+        
+        promiseBlock.done = { obj in
+            XCTFail("unexpected success")
+            
+        }
+        
+        promiseBlock.observed = { obj in
+            XCTFail("unexpected success")
+        }
+        
+        let rejectedExpectation = XCTestExpectation()
+        promiseBlock.rejected = { obj in
+            rejectedExpectation.fulfill()
+        }
+        
+        let alwaysExpect = XCTestExpectation()
+        promiseBlock.always = {
+            alwaysExpect.fulfill()
+        }
+        
+        promiseBlock.call(with: .failure(callObj))
+        
+        self.wait(for: [rejectedExpectation, alwaysExpect], timeout: 10)
+    }
+    
+    func testCallErrorOnQueue() {
+        let promiseBlock = Promise<Void>.PromiseBlock(cancellationToken: PromiseCancelToken())
+        let key = DispatchSpecificKey<String>()
+        promiseBlock.queue = DispatchQueue(label: "testCallNonErrorOnQueue")
+        promiseBlock.queue?.setSpecific(key: key, value: "testCallNonErrorOnQueue")
+        let callObj = DummyError.dummy
+        
+        promiseBlock.done = { obj in
+            XCTFail("unexpected success")
+        }
+        
+        promiseBlock.observed = { obj in
+            XCTFail("unexpected success")
+        }
+        
+        let rejectedExpectation = XCTestExpectation()
+        promiseBlock.rejected = { obj in
+            XCTAssertEqual(DispatchQueue.getSpecific(key: key), "testCallNonErrorOnQueue")
+            rejectedExpectation.fulfill()
+        }
+        
+        let alwaysExpect = XCTestExpectation()
+        promiseBlock.always = {
+            XCTAssertEqual(DispatchQueue.getSpecific(key: key), "testCallNonErrorOnQueue")
+            alwaysExpect.fulfill()
+        }
+        
+        promiseBlock.call(with: .failure(callObj))
+        
+        self.wait(for: [rejectedExpectation, alwaysExpect], timeout: 10)
+    }
```

Comment:
Looks like we can add more tests for validating callback queue when we use a chain of promises.
Case like:
- 1 queue: create promise and callbacks
- dispatch on 2 queue and transfer the promise to it 
- and new callbacks to the promise  
- fullfill promise on 3 queue
- validate